src/main/java/com/hivemq/cli/commands/shell/ShellDisconnectCommand.java

```diff
@@ -94,10 +87,18 @@ public void run() {
         }
 
         try {
-            mqttClientExecutor.disconnect(this);
+            if (disconnectAll) {
+                mqttClientExecutor.disconnectAllClients(this);
+            }
+            else {
+                if (identifier == null) {
+                    Logger.error("Missing required option '--identifier=<identifier>'");
+                }
+                mqttClientExecutor.disconnect(this);
```

Comment:
an error gets thrown but you still do the disconnect action, is this the desired behavior?